birdhouse/components/cowbird/default.env

```diff
@@ -0,0 +1,29 @@
+# -------
+# Cowbird
+# -------
+
+# add any new variables not already in 'VARS' or 'OPTIONAL_VARS' that must be replaced in templates here
+# single quotes are important in below list to keep variable names intact until 'pavics-compose' parses them
+EXTRA_VARS='
+    ${COWBIRD_VERSION}
+    ${COWBIRD_MONGODB_SERVICE}
+    ${COWBIRD_MONGODB_HOST}
+    ${COWBIRD_MONGODB_PORT}
+'
+# extend the original 'VARS' from 'birdhouse/pavics-compose.sh' to employ them for template substitution
+# adding them to 'VARS', they will also be validated in case of override of 'default.env' using 'env.local'
+VARS="$VARS $EXTRA_VARS"
+
+# =====================
+# Cowbird Configuration
+# =====================
+
+export COWBIRD_VERSION="0.5.0"
+
+# reuse the mongo instance provided by the core docker-compose
+# the 'host' is the mongo 'service' inter-docker network connection in this case
+# if another service should be used (e.g.: dedicated mongodb), adjust the variable
+# the selected 'mongo' service should be member of a network of same name
+export COWBIRD_MONGODB_SERVICE=mongodb
```

Comment:
Is this piggyback into the "default" mongodb that is very old?  I remember you had to roll your own mongodb for Weaver so you can have a newer mongodb.  Why not roll your own mongodb container for this component so you have full control on the version again?

Reply:
In this case, it is "ok" to use the old one. The example https://github.com/Ouranosinc/cowbird/blob/master/docker/docker-compose.example.yml uses 3.4 also for testing. In the case of Weaver, there was an actual requirement of 5.0 for a new mongodb feature.

I've actually though about this possibility to have its own mongodb version. This is why I defined this service variable that allows one to override if they want (they would have to provide a `docker-compose.override` with that service as well). Otherwise, I think it is better to avoid wasting more resources for this case.

Reply:
I usually follow the saying "Premature Optimization Is the Root of All Evil". I usually choose ease of maintenance, simplicity and correctness over performance and resource optimization.  An idle mongodb is probably not going to suck that much resources.  But it's your component so your call.